compiler/src/org.graalvm.compiler.replacements/src/org/graalvm/compiler/replacements/IntrinsicGraphBuilder.java

```diff
@@ -127,7 +127,9 @@ protected IntrinsicGraphBuilder(OptionValues options, DebugContext debug, CorePr
     private <T extends ValueNode> void updateLastInstruction(T v) {
         if (v instanceof FixedNode) {
             FixedNode fixedNode = (FixedNode) v;
-            lastInstr.setNext(fixedNode);
+            if (lastInstr != null) {
+              lastInstr.setNext(fixedNode);
+            }
```

Comment:
@dougxc I wonder how this NPE was never a problem in Graal in until now since `lastInstr` can sometimes be `null`.

Reply:
This bug in `IntrinsicGraphBuilder.updateLastInstruction` was exposed when [adding support for compressed pointers](https://github.com/oracle/graal/commit/f478d4a5d92489250bc67bf8e73e5340d3796aa7#diff-ff41b7830c4d716fddac0041450f3a0eR825). Normally `IntrinsicGraphBuilder` has only been used to manually weave a graph that has straight line control flow. Now that it is being used to weave conditional control flow, it is indeed possible for the `lastInstr` to be null. As such, I think the fix in this PR is correct.